tests/unit/stores/UserStore.spec.js

```diff
@@ -0,0 +1,43 @@
+import { setActivePinia, createPinia } from 'pinia';
+import { useUserStore } from 'src/stores/UserStore';
+
+describe('User Store', () => {
+  beforeEach(() => {
+    // creates a fresh pinia and makes it active
+    // so it's automatically picked up by any useStore() call
+    // without having to pass it to it: `useStore(pinia)`
+    setActivePinia(createPinia());
+  });
+
+  it('should test variables in the store', () => {
+    const store = useUserStore();
+
+    expect(store.username).toBe('');
+    expect(store.firstname).toBe('');
+
+    store.username = 'username';
+    store.firstname = 'firstname';
+
+    expect(store.username).toBe('username');
+    expect(store.firstname).toBe('firstname');
+  });
+
+  describe('test store actions', () => {
+    describe('test isEmpty', () => {
+      it('should return true if username and firstname are empty', () => {
+        const store = useUserStore();
```

Comment:
could be externalized outside of the itS

Reply:
Tried it. Unfortunately I can't because we need to set the pinia store in each test using setActivePinia (which is done in the beforeEach)